### ISSUE TITLE
Linked data not shared after export

### ISSUE THREAD

**Issue by Kaesebrot84** [KhronosGroup/glTF-Blender-IO#323]
I am trying to export two objects which are supposed to share mesh data (`NORMAL`, `POSITION`, `indices` etc.). I also want the objects' materials not being shared.

In order to achieve this I linked the objects' meshes via `Link Data`. I also linked the materials to objects instead of linking them to data.

![structure](https://user-images.githubusercontent.com/13865115/53298395-bdc50400-382d-11e9-8f0d-f987b8494cfe.PNG)

![linked_to_object](https://user-images.githubusercontent.com/13865115/53298396-c289b800-382d-11e9-8792-de6c13e48339.PNG)

![pawns](https://user-images.githubusercontent.com/13865115/53298399-c4ec1200-382d-11e9-9a14-c71090f09d86.PNG)

When I export to gltf it seems like according primitives are still pointing to different accessors. I would have expected them to share buffer data.

```json
        {
            "name" : "pawn_1.002",
            "primitives" : [
                {
                    "attributes" : {
                        "NORMAL" : 6,
                        "POSITION" : 4,
                        "TEXCOORD_0" : 5
                    },
                    "indices" : 7,
                    "material" : 1
                }
            ]
        },
        {
            "name" : "pawn_1.002",
            "primitives" : [
                {
                    "attributes" : {
                        "NORMAL" : 10,
                        "POSITION" : 8,
                        "TEXCOORD_0" : 9
                    },
                    "indices" : 11,
                    "material" : 0
                }
            ]
        }
```

The according accessors (`NORMAL` in this case) are also pointing to different `bufferView` elements:

```json
        {
            "bufferView" : 7,
            "componentType" : 5126,
            "count" : 5504,
            "type" : "VEC3"
        },        
        {
            "bufferView" : 12,
            "componentType" : 5126,
            "count" : 5504,
            "type" : "VEC3"
        }
```

How can I make primitives in exported gltf files share buffer information? According to Blender's Add-On screen I am using `Import-Export: glTF 2.0 format, 0.0.1`.

**Comment by donmccurdy** [comment]
When duplicating objects with Alt+D the meshes are reused. I'm not sure what's different in this case – do you get the same result if they share a material?

**Comment by Kaesebrot84** [comment]
> When duplicating objects with Alt+D the meshes are reused. I'm not sure what's different in this case – do you get the same result if they share a material?

Well now I tried duplicating the object via `Alt+D` leaving the underlying material untouched. The results are the same, primitive data still pointing to different accessors and accessors pointing to different `bufferView`s.

**Comment by Cyp** [comment]
If the nodes have modifiers, the meshes are unconditionally duplicated (see #55). Don't remember whether auto-smooth counts as a modifier for that, but think so.

**Comment by UX3D-schmithuesen** [comment]
You can't have two different materials used on a shared primitive (=linked duplicate) because the primitive references its material.

**Comment by emackey** [comment]
We could re-use accessors though.  That would be preferable to duplicated mesh data.

**Comment by Kaesebrot84** [comment]
> You can't have two different materials used on a shared primitive (=linked duplicate) because the primitive references its material.

As I mentioned before, it doesn't work with both primitives having the same material (at least in my case). What is exactly the expected behavior? Should it be possible to share data between primitives at all?

**Comment by donmccurdy** [comment]
Let's leave this open as a feature request rather than a bug. I agree that reusing accessor data would be a good solution.

**Comment by UX3D-schmithuesen** [comment]
> As I mentioned before, it doesn't work with both primitives having the same material either (at least in my case). What is exactly the expected behavior? Should it be possible to share data between primitives at all? If not, wouldn't that be a convenient way to save a lot of space?

Oh! I didn't notice this somehow. Sharing primitives works fine for me, here is how I tested it:
* create a new object with some mesh (I just use the built in monkey “Suzanne”)
* add a material
* use `Object -> Duplicate Linked` (Shift + D) to create a duplicate
* export

there are two nodes, but just one mesh/material:
```
"nodes" : [
    {
        "mesh" : 0,
        "name" : "Suzanne"
    },
    {
        "mesh" : 0,
        "name" : "Suzanne.001",
        "translation" : [
            0.9270802736282349,
            0.37051883339881897,
            -2.5556652545928955
        ]
    }
],
"materials" : [
    {
        "name" : "Material.002",
        "pbrMetallicRoughness" : {
            "baseColorTexture" : {
                "index" : 0,
                "texCoord" : 0
            },
            "metallicFactor" : 0,
            "roughnessFactor" : 0.5
        }
    }
],
"meshes" : [
    {
        "name" : "Suzanne",
        "primitives" : [
            {
                "attributes" : {
                    "POSITION" : 0,
                    "NORMAL" : 1,
                    "TEXCOORD_0" : 2
                },
                "indices" : 3,
                "material" : 0
            }
        ]
    }
],
```

**Comment by Kaesebrot84** [comment]
@UX3D-schmithuesen I tried your solution a couple of times. In my case it did work when using `Object -> Duplicate Linked`. It did not work when using the shortcut `Shift+D`, though.

**Comment by UX3D-schmithuesen** [comment]
Maybe the shortcut is different on your platform, but `Duplicate Linked` is what you need.

**Comment by Kaesebrot84** [comment]
> Maybe the shortcut is different on your platform, but `Duplicate Linked` is what you need.

So I played around with this functionality a little and I had a weird behavior. I have two cases in the first case I have a blender file with the monkey mesh only. 

1. I apply `Duplicate Linked` and export it to gltf. The result looks as expected, there are two nodes but only one mesh.

![working_duplication](https://user-images.githubusercontent.com/13865115/54026718-205bbf80-419f-11e9-836e-becfe147e30a.gif)

2. In the second case I have another scene already set up. I add a monkey object, apply `Duplicate Linked` and export it to gltf. This time two different meshes are used for the supposedly linked data.

![failing_duplication](https://user-images.githubusercontent.com/13865115/54026861-a7109c80-419f-11e9-81b8-936ed55670d3.gif)
Did I miss something? Or is there some requirement I am not considering in the second case?

**Comment by Taroslord** [comment]
Is this issue fixed? I've a similar problem with linked scenes in Blender 2.9x.

I've the a model used in two scenes (linked). The model won't be exported until I delete the it in the second scene or the whole second scene, where the model is linked in.

Is this the same issue?

**Comment by donmccurdy** [comment]
@Taroslord if the model is failing to export entirely, I think it is a different/unknown issue. Could you file an issue and include a `.blend` file?

**Comment by Taroslord** [comment]
Sure.

I've reproduced my scene to an neutral example, where the problem appears. See my .blend file below to reproduce the problem.

**Read this first:**
The example model is animated and has two parent empties, like in my original scene.
There is a second scene created. It is called _Scene_Linked_. The content is linked with the scene called: _Scene_.

**How to reproduce the bug:**
1. Select the "root_Torus" and its child objects in the outliner window. (Select Hierarchy)
2. Export to .glb
-> Error appears. Export impossible.

**To avoid the bug:**
1. Delete "Scene_Linked" completely.
2. Select the "root_Torus" and its child objects in the outliner window. (Select Hierarchy)
3. Export to .glb
-> All works.

**What I expect:**
I want to use linked scenes in my Blend files and keep the ability to export my object hierarchies to _.glb_ files, even if they are linked within other scenes.

Thank you for your time.

[cw_LinkedScene_ExportBug.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/5372674/cw_LinkedScene_ExportBug.zip)

**Comment by donmccurdy** [comment]
Thanks! I've copied these details to https://github.com/KhronosGroup/glTF-Blender-IO/issues/1236 so we can track it.

**Comment by julienduroure** [comment]
Seems to be fixed by #1378

**Comment by julienduroure** [comment]
Fixed by 69746b2b31eeee080d5dfd3c74f1e02742795ed2
